# Numpy 


In [1]:
import pandas as pd
import numpy as np
import multiprocessing as mp
from time import time

In [2]:
time1 = time()
random=np.random.normal(10,0.1,(20000,20000))
# random=np.random.randint(0,10,(6,6))
df=pd.DataFrame(random,index=np.arange(20000), columns=np.arange(20000))
print(np.mean(df[::100],axis=0))

print('\nPool Parallelism runs %0.3f seconds using numpy\n' % ((time()-time1)))

0        10.009633
1         9.997004
2        10.001726
3        10.014165
4        10.004132
           ...    
19995    10.000295
19996     9.999984
19997     9.996283
19998     9.988582
19999    10.004378
Length: 20000, dtype: float64

Pool Parallelism runs 15.608 seconds using numpy



# Dask

In [3]:
import dask.array as da

In [4]:
time1 = time()
daskrandom=da.random.normal(10,0.1,size=(20000,20000),chunks=(1000, 1000))
daskmean = daskrandom[::100].mean(axis=0)
print(daskmean.compute())
print('\nPool Parallelism runs %0.3f seconds using Dask\n' % ((time()-time1)))

[ 9.99972989  9.9919314   9.98520677 ... 10.00611504  9.99929536
  9.99795468]

Pool Parallelism runs 4.019 seconds using Dask



In [5]:
daskrandom

dask.array<normal, shape=(20000, 20000), dtype=float64, chunksize=(1000, 1000), chunktype=numpy.ndarray>

In [6]:
daskmean

dask.array<mean_agg-aggregate, shape=(20000,), dtype=float64, chunksize=(1000,), chunktype=numpy.ndarray>

# Performance comparision:
Time taken by numpy array is 15.608 seconds
Time taken by dask array is 4.019 seconds
Numpy took 16 seconds which is 4 time the array